infrastructure/network/addressmanager/addressmanager.go

```diff
@@ -5,1585 +5,225 @@
+	"github.com/pkg/errors"
 )
 
-// AddressKey represents a "string" key in the form of ip:port for IPv4 addresses
-// or [ip]:port for IPv6 addresses for use as keys in maps.
-type AddressKey string
-type newAddressBucketArray [NewBucketCount]map[AddressKey]*KnownAddress
-type triedAddressBucketArray [TriedBucketCount][]*KnownAddress
-
-// AddressManager provides a concurrency safe address manager for caching potential
-// peers on the Kaspa network.
-type AddressManager struct {
-	cfg             *config.Config
-	databaseContext *dbaccess.DatabaseContext
-
-	mutex              sync.Mutex
-	lookupFunc         func(string) ([]net.IP, error)
-	random             *rand.Rand
-	key                [32]byte
-	addressIndex       map[AddressKey]*KnownAddress // address keys to known addresses for all addresses.
-	started            int32
-	shutdown           int32
-	wg                 sync.WaitGroup
-	quit               chan struct{}
-	localAddressesLock sync.Mutex
-	localAddresses     map[AddressKey]*localAddress
-	localSubnetworkID  *subnetworkid.SubnetworkID
-
-	fullNodeNewAddressBucketArray     newAddressBucketArray
-	fullNodeNewAddressCount           int
-	fullNodeTriedAddressBucketArray   triedAddressBucketArray
-	fullNodeTriedAddressCount         int
-	subnetworkNewAddressBucketArrays  map[subnetworkid.SubnetworkID]*newAddressBucketArray
-	subnetworkNewAddressCounts        map[subnetworkid.SubnetworkID]int
-	subnetworkTriedAddresBucketArrays map[subnetworkid.SubnetworkID]*triedAddressBucketArray
-	subnetworkTriedAddressCounts      map[subnetworkid.SubnetworkID]int
-}
-
-type serializedKnownAddress struct {
-	Address       AddressKey
-	SourceAddress AddressKey
-	SubnetworkID  string
-	Attempts      int
-	TimeStamp     int64
-	LastAttempt   int64
-	LastSuccess   int64
-	IsBanned      bool
-	BannedTime    int64
-	// no refcount or tried, that is available from context.
+// AddressRandomizer is the interface for the randomizer needed for the AddressManager.
+type AddressRandomizer interface {
+	RandomAddress(addresses []*appmessage.NetAddress) *appmessage.NetAddress
+	RandomAddresses(addresses []*appmessage.NetAddress, count int) []*appmessage.NetAddress
 }
 
-type serializedNewAddressBucketArray [NewBucketCount][]AddressKey
-type serializedTriedAddressBucketArray [TriedBucketCount][]AddressKey
-
-// PeersStateForSerialization is the data model that is used to
-// serialize the peers state to any encoding.
-type PeersStateForSerialization struct {
-	Version   int
-	Key       [32]byte
-	Addresses []*serializedKnownAddress
-
-	SubnetworkNewAddressBucketArrays   map[string]*serializedNewAddressBucketArray // string is Subnetwork ID
-	FullNodeNewAddressBucketArray      serializedNewAddressBucketArray
-	SubnetworkTriedAddressBucketArrays map[string]*serializedTriedAddressBucketArray // string is Subnetwork ID
-	FullNodeTriedAddressBucketArray    serializedTriedAddressBucketArray
-}
-
-type localAddress struct {
+type addressEntry struct {
```

Comment:
If we only have netAddress inside addressEntry, do we really need a especial struct for it? 
Why not store appmessage.NetAddress directly?